src/neuroglancer/widget/segment_set_widget.ts

```diff
@@ -66,6 +66,43 @@ export class SegmentSetWidget extends RefCounted {
     this.registerDisposer(displayState.segmentColorHash.changed.add(() => {
       this.handleColorChanged();
     }));
+    this.registerDisposer(displayState.segmentSelectionState.changed.add(() => {
+      const segmentID = this.segmentSelectionState.selectedSegment.toString();
+      const segmentButton = <HTMLElement>document.querySelector(`[data-segID="${segmentID}"]`);
+      const existingHlight = document.querySelectorAll(`.segSelected`);
```

Comment:
No need to scan the whole `document` - `element` should contain all the buttons you want to change. Also reduces the chance for side effects.

Reply:
Another thing: `getElementsByClassName('segSelected')` is more specific/readable than `querySelectorAll`